### PR TITLE
Unity service example

### PR DESCRIPTION
**Pull request by hyounesy**
Example code and tutorial for UnityService

### CODE REVIEW

File: tutorials/ros_packages/robotics_demo/scripts/server_endpoint.py [hunk 1]
```diff
@@ -16,7 +16,8 @@ def main():
     tcp_server.start({
         'pos_srv': RosService('position_service', PositionService),
```
Comment (LaurieCheers-unity) [inline tutorials/ros_packages/robotics_demo/scripts/server_endpoint.py:17]: Based on our conversation, should we change one of these strings?

Comment (hyounesy) [reply tutorials/ros_packages/robotics_demo/scripts/server_endpoint.py:17]: I think that is a good idea to change it to be both `pos_srv`. Should it be done in a separate PR?

File: tutorials/ros_unity_integration/README.md [hunk 2]
```diff
@@ -26,8 +27,12 @@ Example scripts implemented in tutorials:
...
+- `unity_scripts/RosSubscriberExample.cs`
+	- Subscribes to a topic that accepts color messages and uses them to change the color of a GameObject in the Unity scene.
+
 - `unity_scripts/RosServiceExample.cs`
 	- Returns a destination position for a GameObject to move towards each time the service is called.
 
-- `unity_scripts/RosSubscriberExample.cs`
-	- Subscribes to a topic that accepts color messages and uses them to change the color of a GameObject in the Unity scene.
+- `unity_scripts/RosUnityServiceExample.cs`
+	- Runs a service in the Unity scene that can takes GameObject names in a request and responds with the GameObject's Pose.
```
Comment (LaurieCheers-unity) [inline tutorials/ros_unity_integration/README.md:37]: Grammar "takes" -> "take"

How about, "Runs a service in the Unity scene that takes a GameObject's name and responds with the Pose of that object."

File: tutorials/ros_unity_integration/unity_scripts/RosUnityServiceExample.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+using UnityEngine;
+using Unity.Robotics.ROSTCPConnector;
+using Unity.Robotics.ROSTCPConnector.ROSGeometry;
+
+/// <summary>
+/// Example demonstration of implementing a UnityService that receives a Request message from another ROS node and sends a Response back
+/// </summary>
+public class RosUnityServiceExample : MonoBehaviour
+{
+    public string serviceName = "obj_pose_srv";
```
Comment (LaurieCheers-unity) [inline tutorials/ros_unity_integration/unity_scripts/RosUnityServiceExample.cs:11]: Coding standards: this should be a const named k_ServiceName

File: tutorials/ros_packages/robotics_demo/scripts/object_pose_client.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+def usage():
+    return "%s [object_name]"%sys.argv[0]
+
+if __name__ == "__main__":
+    if len(sys.argv) == 2:
+        name = str(sys.argv[1])
+    else:
+        print(usage())
+        sys.exit(1)
+    print("Requestinng pose for %s"%(name))
```
Comment (LaurieCheers-unity) [inline tutorials/ros_packages/robotics_demo/scripts/object_pose_client.py:31]: Single n in "requesting"

File: tutorials/ros_unity_integration/unity_scripts/RosUnityServiceExample.cs (hunk 3 above)
Comment (hyounesy) [reply tutorials/ros_unity_integration/unity_scripts/RosUnityServiceExample.cs:11]: hmmm... should it really? This is basically similar to the topic/service name in the other 3 examples, which are defined as public properties.

File: tutorials/ros_packages/robotics_demo/scripts/server_endpoint.py (hunk 1 above)
Comment (LaurieCheers-unity) [reply tutorials/ros_packages/robotics_demo/scripts/server_endpoint.py:17]: I don't feel like that's necessary. Just test that the service tutorial still works.

File: tutorials/ros_unity_integration/unity_scripts/RosUnityServiceExample.cs (hunk 3 above)
Comment (LaurieCheers-unity) [reply tutorials/ros_unity_integration/unity_scripts/RosUnityServiceExample.cs:11]: The other examples are not written according to coding standards, we need to update them at some point :-)

Comment (LaurieCheers-unity) [reply tutorials/ros_unity_integration/unity_scripts/RosUnityServiceExample.cs:11]: Oh, I guess the idea is for this to be editable in the inspector... ok, in that case I believe it should be

[SerializeField]
string m_ServiceName = "obj_pose_srv";

File: tutorials/ros_packages/robotics_demo/scripts/object_pose_client.py (hunk 4 above)
Comment (hyounesy) [reply tutorials/ros_packages/robotics_demo/scripts/object_pose_client.py]: good catch. fixed 7a59a02

File: tutorials/ros_unity_integration/unity_scripts/RosUnityServiceExample.cs (hunk 3 above)
Comment (hyounesy) [reply tutorials/ros_unity_integration/unity_scripts/RosUnityServiceExample.cs]: fixed 7a59a02

File: tutorials/ros_unity_integration/README.md (hunk 2 above)
Comment (hyounesy) [reply tutorials/ros_unity_integration/README.md]: good catch. fixed 7a59a02

File: tutorials/ros_packages/robotics_demo/scripts/server_endpoint.py (hunk 1 above)
Comment (hyounesy) [reply tutorials/ros_packages/robotics_demo/scripts/server_endpoint.py]: done 7b4cd05

**Review by LaurieCheers-unity — APPROVED** [review]

**Pull request merged**